### DISCUSSION TITLE
Javascript autocompletition from java classes.

### DISCUSSION THREAD

**Discussion by JuliGame** [mozilla/rhino#1323] [Q&A] [answered]
I am using rhino in Java to run some js scripts. This scripts have access to my Java classes and they create new objects from them.
I want to know if there is an easy way of telling my IDE that this objects in js are really Java objects so that I can auto complete and have snippets. It would make coding this scripts a lot faster.

**Comment by p-bakker** [comment]
That really depends the ide you use, but I doubt there are many (if any) that will do that for you.

If you use an ide that supports typescript, you might be able to provide d.ts with all the typings of the java stuff you use

**Comment by JuliGame** [comment, marked as answer]
Right now using https://github.com/bsorrentino/java2typescript
Its working fine, really good tool!
